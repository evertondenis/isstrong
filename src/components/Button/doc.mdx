---
name: Button
menu: Components
---

import { Playground, PropsTable } from 'docz'
import Button from './'

# Button

<PropsTable of={Button} />

## Basic usage

<Playground>
  <Button>Click Me</Button>
  <Button disabled>Click Me</Button>
</Playground>
