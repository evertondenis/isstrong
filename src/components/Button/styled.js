import styled from 'styled-components'

export const StyledButton = styled.button`
  display: inline-block;
  margin: 25px 0;
  padding: 10px;
  width: 100%;
  -webkit-user-select: none;
  border: 0;
  background-image: none;
  background: #17D499;
  cursor: pointer;
  font-size: 18px;
  font-weight: normal;
  color: #fff;
  line-height: 1.5;
  text-align: center;
  transition: all .2s linear;
  white-space: nowrap;

  span {
    color: #fff;
    text-decoration: none;
  }

  &:focus {
    background: #17D499;
    color: #fff;

    span {
      color: #fff;
    }
  }

  &:hover:not([disabled]) {
    color: #fff;
    background-color: #17D499;
    transition: all .2s linear;

    span {
      color: #FFF;
      text-decoration: none;
    }
  }

  &:disabled {
    color: #efefef;
    background: #ccc;
    cursor: not-allowed;
  }
`
