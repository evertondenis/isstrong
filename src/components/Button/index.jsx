import React from 'react'
import PropTypes from 'prop-types'
import { StyledButton } from './styled'

const Button = (props) => {
  const { type, onClick, disabled, className, form, children } = props

  return (
    <StyledButton
      type={type}
      onClick={!disabled ? onClick : undefined}
      disabled={disabled}
      className={className}
      form={form}
    >
      {children}
    </StyledButton>
  )
}


Button.defaultProps = {
  children: '',
  className: '',
  disabled: '',
  form: 'form',
  onClick: null,
  type: 'button'
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  disabled: PropTypes.any,
  form: PropTypes.string,
  onClick: PropTypes.func,
  type: PropTypes.string,
}

export default Button
