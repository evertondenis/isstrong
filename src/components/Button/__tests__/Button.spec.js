import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import Button from '../'
import 'setupTest'

describe('Button.', () => {
  it('Should be render <Button />.', () => {
    const wrapper = shallow(
      <Button>Send</Button>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('Should be render disabled <Button />.', () => {
    const wrapper = shallow(
      <Button disabled>Send</Button>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should be clicked', () => {
    const onClick = jest.fn()
    const component = shallow(
      <Button onClick={onClick}>
        Content
      </Button>
    )
    component.simulate('click')
    expect(onClick).toHaveBeenCalled()
  })
})
