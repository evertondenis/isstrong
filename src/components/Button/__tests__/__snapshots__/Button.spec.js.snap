// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button. Should be render <Button />. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    className=""
    disabled=""
    form="form"
    onClick={null}
    type="button"
>
    Send
</Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Send",
      "className": "",
      "disabled": "",
      "form": "form",
      "onClick": null,
      "type": "button",
    },
    "ref": null,
    "rendered": "Send",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Send",
        "className": "",
        "disabled": "",
        "form": "form",
        "onClick": null,
        "type": "button",
      },
      "ref": null,
      "rendered": "Send",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Button. Should be render disabled <Button />. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    className=""
    disabled={true}
    form="form"
    onClick={null}
    type="button"
>
    Send
</Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Send",
      "className": "",
      "disabled": true,
      "form": "form",
      "onClick": undefined,
      "type": "button",
    },
    "ref": null,
    "rendered": "Send",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Send",
        "className": "",
        "disabled": true,
        "form": "form",
        "onClick": undefined,
        "type": "button",
      },
      "ref": null,
      "rendered": "Send",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
